queue/queue.go

```diff
@@ -6,15 +6,28 @@ import (
 	"time"
 
 	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/config"
 	"github.com/frain-dev/convoy/datastore"
+	"github.com/go-redis/redis/v8"
+	"github.com/vmihailenco/taskq/v3"
 )
 
 type Queuer interface {
 	io.Closer
 	Write(context.Context, convoy.TaskName, *datastore.EventDelivery, time.Duration) error
 }
 
+type QueueClient interface {
+	NewClient(config.Configuration) (*StorageClient, taskq.Factory, error)
+	NewQueue(StorageClient, taskq.Factory, string) Queuer
+}
+
 type Job struct {
 	Err error  `json:"err"`
 	ID  string `json:"id"`
 }
+
+type StorageClient struct {
```

Comment:
@ogbanugot What we will rather do here is to create our own `QueueOptions`. This will contain a subset of fields from `taskq.QueueOptions` and will be passed into `NewQueue`.

Reply:
Can QueueOptions be defined here though? It will cause an import cycle.

Reply:
Also, I think `memqueue` and `redisqueue` can both return an instance of our own `QueueOptions` so that we can toggle between the two in `cmd/main.go` with ease.